Userland/Libraries/LibWeb/WebDriver/ExecuteScript.cpp

```diff
@@ -331,20 +331,22 @@ ExecuteScriptResultSerialized execute_async_script(Web::Page& page, DeprecatedSt
     auto& vm = window->vm();
     auto start = MonotonicTime::now();
 
+    // NOTE: We need to push an execution context in order to make create_resolving_functions() succeed.
+    vm.push_execution_context(settings_object.realm_execution_context());
+    ScopeGuard pop_guard = [&] {
+        VERIFY(&settings_object.realm_execution_context() == &vm.running_execution_context());
+        vm.pop_execution_context();
+    };
+
     // 4. Let promise be a new Promise.
-    auto promise = JS::Promise::create(realm);
+    auto promise_capability = WebIDL::create_promise(realm);
+    JS::NonnullGCPtr promise { verify_cast<JS::Promise>(*promise_capability->promise()) };
 
     // FIXME: 5 Run the following substeps in parallel:
-    auto result = [&] {
-        // NOTE: We need to push an execution context in order to make create_resolving_functions() succeed.
-        vm.push_execution_context(settings_object.realm_execution_context());
-
+    [&] {
```

Comment:
What is the purpose of this being a lambda? :thonk:

Reply:
I left it to be lambda because "then abort these steps" in the spec can be nicely translated to early return from lambda :) also eventually it will have to be lambda because says spec that it should run in parallel.